syntax = "proto3";

package storage;
option go_package = "github.com/oasislabs/ekiden/go/grpc/storage";

service Storage {
    rpc Get(GetRequest) returns (GetResponse) {}
    rpc Insert(InsertRequest) returns (InsertResponse) {}
    rpc GetKeys(GetKeysRequest) returns (GetKeysResponse) {}
}

message GetRequest {
    bytes id = 1;
}

message GetResponse {
    bytes data = 1;
}

message InsertRequest {
    bytes data = 1;
    uint64 expiry = 2;
}

message InsertResponse {
}

message GetKeysRequest {
}

message GetKeysResponse {
    repeated bytes keys = 1;
    repeated uint64 expiry = 2;
}
